### ISSUE TITLE
process_raw! does not accept SeisChannel's 

### ISSUE THREAD

**Issue by Julians42** [tclements/SeisNoise.jl#45]
Process_raw does not seem to handle SeisData objects with multiple channels, only SeisData objects with single channels (see screenshot). Might be a nice feature instead of converting to an array of single channel SeisData objects. Broadcasting would also be a nice feature, but more involved. 

<img width="1023" alt="Screen Shot 2020-06-30 at 9 29 10 AM" src="https://user-images.githubusercontent.com/43148195/86131875-31c38c00-bab4-11ea-9b33-a7bb0684ddd7.png">

**Comment by tclements** [comment]
Just confirmed this is failing on v0.4.0:

```julia
(@v1.4) pkg> st
Status `~/.julia/environments/v1.4/Project.toml`
  [b372bb87] SeisIO v1.1.0 #master (https://github.com/jpjones76/SeisIO.jl.git)
  [8cc7c3c0] SeisNoise v0.4.0
```

```julia
using SeisNoise, SeisIO
C = SeisIO.RandSeis.randSeisChannel(s=true,c=false,fs_min=20.)
ungap!(C)
process_raw!(C,10.)
ERROR: MethodError: no method matching process_raw!(::SeisChannel, ::Float64)
Closest candidates are:
  process_raw!(::SeisData, ::Float64; ϕshift) at /home/timclements/.julia/packages/SeisNoise/I9Knu/src/compute_fft.jl:21
Stacktrace:
 [1] top-level scope at none:0
```

but on the master branch it works:

```julia
(@v1.4) pkg> st
Status `~/.julia/environments/v1.4/Project.toml`
  [b372bb87] SeisIO v1.1.0 #master (https://github.com/jpjones76/SeisIO.jl.git)
  [8cc7c3c0] SeisNoise v0.4.1 #master (https://github.com/tclements/SeisNoise.jl.git)
 ```

```julia
using SeisNoise, SeisIO
C = SeisIO.RandSeis.randSeisChannel(s=true,c=false,fs_min=20.)
ungap!(C)
process_raw!(C,10.)

# test with Array{SeisData,1}
ar = [SeisIO.RandSeis.randSeisData(s=1.,c=0.,fs_min=10.) for _ = 1:5]
process_raw!.(ar,10.)
all(vcat([ar[ii].fs .== 10. for ii = 1:length(ar)]...))
true
```

I tagged a new version of SeisNoise (v0.4.1) that has these changes. I will add it to the general registry once this build to passes (https://travis-ci.org/github/tclements/SeisNoise.jl/builds/703539782) ~ 1 hour. For now you should be able to use the master branch:

```julia
pkg> add SeisNoise#master
```

**Comment by tclements** [comment]
Found there is an issue with resampling gapless `SeisChannel` https://github.com/jpjones76/SeisIO.jl/issues/51. This will give a problem with creating `RawData` from `SeisChannel` right now. I'll try to get a patch soon.

**Comment by tclements** [comment]
Fixed by #50
